src/jdk.jartool/share/classes/sun/tools/jar/resources/jar.properties

```
@@ -90,6 +90,8 @@ error.release.value.toosmall=\
         release {0} not valid, must be >= 9
 error.release.unexpected.versioned.entry=\
         unexpected versioned entry {0} for release {1}
+error.release.value.toohigh.versioned.entry=\
+        classfile release value of {0} too high: {1}
```

Not sure about the term 'release value' here. Is that defined somewhere?

Maybe leaning on 'class file major version' like this would work better?

```suggestion
error.release.value.toohigh.versioned.entry=\
        class file {0} targeting version '{1}' must have a major version lower or equal to '{2}' , but was '{3}'
```

And then `{0}` is the entry, `{1}` is the `mrversion`, `{2}` is the class file major version associated with that `mrversion`, and `{3}` was the actual class file major version.

(also, 'class file' is two words)

---

In this context the actual class file major version doesn't add any value to communicate the issue to the user:
- `javac --release N` and
- `jar --release N` use the "Java release feature number" in their options, so does the
- `META-INF/versions/N` path.

---

User-facing help messages read:
```
javac --help
...
  --release <release>
        Compile for the specified Java SE release.
        Supported releases:
            8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25
```
```
jar --help
...
      --release VERSION      Places all following files in a versioned directory
                             of the jar (i.e. META-INF/versions/VERSION/)
```

---

Maybe this reads better? `release version of {0} too high: {1}`:
```
release version of META-INF/versions/9/version/Version.class too high: 25
release version of META-INF/versions/10/version/Version.class too high: 25
```
Note that the path of the JAR entry contains the target release version number, and `Version.class` was compiled with `javac` 25 (with `--release 25`).

---

Prior-art in the messages do read:
```properties
error.release.value.notnumber=\
        release {0} not valid
error.release.value.toosmall=\
        release {0} not valid, must be >= 9
error.release.unexpected.versioned.entry=\
        unexpected versioned entry {0} for release {1}
```

---

According to above's prior-art, it could just read `release of {0} too high: {1}`

---

> `release of {0} too high: {1}`

I think this is better. Probably good enough.

My concern stems from the fact that 'release' doesn't seem to be defined for _class files_. I see that the term 'release' exists, and maps to a versioned directory, or you can compile classes _for_ a certain release, But, it seems that 'the release of a class file' is a new concept here. (after all, `javac` doesn't 'produce class files _with_ a certain release).

Maybe: `class file {fp.entryName()} compiled for release {fp.classReleaseVersion()}, is illegal in versioned directory for version {fp.mrversion()}`

---

Let's try it out with the two examples from the initial description:

```
class file META-INF/versions/9/version/Version.class compiled for release 25, is illegal in version directory 9
class file META-INF/versions/10/version/Version.class compiled for release 25, is illegal in version directory 10
```

---

What about:

```
META-INF/versions/9/version/Version.class of class file major version 69 is unsupported in versioned release 9; only supported in release >= 25
META-INF/versions/10/version/Version.class of class file major version 69 is unsupported in versioned release 10; only supported in release >= 25
```